core/schema/directory.go

```diff
@@ -352,26 +354,105 @@ type dirDockerBuildArgs struct {
 	Secrets    []core.SecretID                    `default:"[]"`
 }
 
-func (s *directorySchema) dockerBuild(ctx context.Context, parent *core.Directory, args dirDockerBuildArgs) (*core.Container, error) {
-	platform := parent.Query.Platform()
+func getDockerIgnoreFileContent(ctx context.Context, parent dagql.Instance[*core.Directory], filename string) ([]byte, error) {
+	file, err := parent.Self.File(ctx, filename)
+	if err != nil {
+		if strings.Contains(err.Error(), "no such file or directory") {
+			return nil, nil
+		}
```

Comment:
`errors.Is()` please! Manual comparison of error strings is very fragile.

Reply:
I don't think we have that here. This error is a custom error coming from here: https://github.com/dagger/dagger/blob/main/core/directory.go#L238

Reply:
fair enough - we should update that error then.

Reply:
👍, I had to return `syscall.ENOENT` to ensure the error returned is still "no such file or directory", otherwise this would become a backward incompatible change.